### ISSUE TITLE
Some users are unable to make towns.

### ISSUE THREAD

**Issue by Flenix** [MyEssentials/MyTown2#289]
Hello.

This has been an ongoing issue for me - for a few months now. Recently I've really been trying to get it sorted though as it is having an impact on my new players.

I was originally using Sqlite for my data storage, and noticed every time someone tried to make a town it would spam the console with database locks issues. The town would be created but under the name SERVER ADMINS, and the resident wouldn't own the town. The resident would get a message stating the plot they were in was already claimed - /t map shows no areas nearby claimed at all.

I asked on IRC and AfterWind told me it was a TPS related issue. However, I logged the TPS for a while and it stayed at a flat 20, until someone did a town command where the DB would lock and the TPS would fall to around 12 for a few minutes, then return to normal.

After more discussion on IRC, I was told that switching to MySQL would solve the issue. I attempted to convert my database but couldn't get it working (Not very good with DB stuff), so ended up just creating an entirely new MySQL and recreating towns.

This didn't fix my issue at all - newbies who try and make town still often get the same issue. However, there's nothing in the console about a database lock, and the TPS is totally unaffected. Been trying to ask on IRC for a few days but haven't got a reply.

TL;DR. Users can't make towns sometimes, Using MySQL, TPS is fine
http://ctrlv.in/622153

(Note I deleted most of the SERVER ADMINS towns just left that as an example.)

**Comment by AfterWind** [comment]
What exactly are you using? Cauldron? Bukkit? I can't think of anything else, unless is a database error in which case: have you got any errors in the logs? Maybe some related to the database?

**Comment by Flenix** [comment]
Forge/ForgeEssentials. There's no error in the console from what I see but I'll search the logs properly in just a moment.

The database was new over the weekend. I can get a copy if you think it might help? It's the only one on there so it's not like there's something else trying to use MySQL or anything.

**Comment by AfterWind** [comment]
"The town would be created but under the name SERVER ADMINS, and the resident wouldn't own the town. The resident would get a message stating the plot they were in was already claimed"
The two can't happen at the same time. Either the town is created without any residents or the town is prevented from being created because the chunk was already claimed.

Are you using latest versions of everything?

**Comment by Flenix** [comment]
Right, digging through the logs, I found that the SERVER ADMIN towns don't seem to create correctly.

For example, one of my regular users remade their town after the new database, and I got this on the console when they did:

[17:07:01] [Server thread/DEBUG] [MyTown2/]: Saving Town EouEou_Town
[17:07:01] [Server thread/DEBUG] [MyTown2/]: Saving TownBlock 0;21;605
[17:07:01] [Server thread/DEBUG] [MyTown2/]: Saving Flag MOBS for Town EouEou_Town
[17:07:01] [Server thread/DEBUG] [MyTown2/]: Saving Flag FAKERS for Town EouEou_Town
[17:07:01] [Server thread/DEBUG] [MyTown2/]: Saving Flag MODIFY for Town EouEou_Town
[17:07:01] [Server thread/DEBUG] [MyTown2/]: Saving Flag RESTRICTIONS for Town EouEou_Town
[17:07:01] [Server thread/DEBUG] [MyTown2/]: Saving Flag NEARBY for Town EouEou_Town
[17:07:01] [Server thread/DEBUG] [MyTown2/]: Saving Rank EouEou_Town;Resident
[17:07:02] [Server thread/DEBUG] [MyTown2/]: Saving Rank EouEou_Town;Mayor
[17:07:03] [Server thread/DEBUG] [MyTown2/]: Saving Rank EouEou_Town;Assistant

However, when I searched for AfterVille (screenshot example above), I only get a single line:
[06:21:24] [Server thread/DEBUG] [MyTown2/]: Saving Town AfterVille
No stack trace, nothing related either side. Obviously I have no way to see who created the town either.

Version wise, I'm using:
MyTown2-1.7.10-219-dev
MyEssentials-Core-1.7.10-52-dev
MyPermissions-1.7.10-13-dev

And also relevant:
ForgeEssentials-1.7.10-1.4.2.863
Forge 1420 (Restricted from recommended due to IC2 not updating to it and crashing me if I try)

**Comment by AfterWind** [comment]
I'll have to dig more into this. So far I have no idea what might cause this.

**Comment by Flenix** [comment]
Alright, let me know if you need anything more from me

**Comment by AfterWind** [comment]
Try latest build, I fixed something related to saving residents in the database, that could be the cause for this.

**Comment by AfterWind** [comment]
Closing. Open if it occurs again.
